[Core] [Unit Test] Core module unit tests take very long time due to heavy GC activity #1544

vladbailescu opened the pull request:
* Created extension for clearing out self4j-test loggers

Closes #1544

bpauli reviewed (commented):
This approach is not thread safe. If tests are running in parallel the memory state can be cleared by another thread which leads to failing tests.

vladbailescu commented:
@bpauli , you're right, that was not a threadsafe approach (but calling `clear` was not working). I removed the `uk.org.lidalia.slf4j-test` and replaced it with Mockito verification.
